automation/taskcluster/decision_task.py

```diff
@@ -61,23 +61,8 @@ def raptor(builder, is_staging):
     build_task = builder.craft_assemble_raptor_task(variant)
     signing_task = builder.craft_raptor_signing_task(build_task['label'], variant, is_staging)
 
-    tasks = [build_task, signing_task]
-
-    for abi in ('armeabi-v7a', 'arm64-v8a'):
-        variant_apk = variant.get_apk(abi)
-        all_raptor_craft_functions = [
-            builder.craft_raptor_tp6m_cold_task(for_suite=i)
-                for i in range(1, 28)
-            ] + [
-                builder.craft_raptor_youtube_playback_task,
-            ]
-        for craft_function in all_raptor_craft_functions:
-            raptor_task = craft_function(
-                signing_task['label'], mozharness_task_id, variant_apk, gecko_revision, is_staging
-            )
-            tasks.append(raptor_task)
-
-    return tasks
+    # Raptor tests are generated in taskgraph
+    return [build_task, signing_task]
```

Comment:
I'm glad to explain 🙂 Here's what happens:

1. taskgraph starts and looks in every `kind.yml` what tasks to generate first. In our case the first kind is `old-decision` because every other task says they depend on it.
2. taskgraph then runs the loader of `old-decision`. This loader is the special shim I wrote in #5257.
    a. Back in #5257, the loader delegated the creation of *all* tasks to `decision_task.py`. Once created, the loader returned all tasks verbatim, and taskgraph just did a few conversions (like labels -> `taskId`s) and schedule them
    b. Now, the loader delegates the creation of the build and the signing tasks `decision_task.py`. 
3. Now, taskgraph takes the build and signing tasks and puts them all under the "old-decision" kind, which is understood by the rest of taskgraph, including the raptor kind.
4. The raptor kind knows which task it depends on thanks to the dep label. 

That's how taskgraph knows where the build and the signing tasks are. old-decision loader acts as a proxy for the task creation, and taskgraph knows everything about what old-decision creates, including task labels.

Please let me know if you need more info :)

Reply:
I think that make sense, thanks!